src/util-napatech.c

```diff
@@ -956,78 +1481,380 @@ uint32_t NapatechSetupTraffic(uint32_t first_stream, uint32_t last_stream,
             exit(EXIT_FAILURE);
         }
 
-        uint8_t start, end;
-        if (strncmp(port->val, "all", 3) == 0) {
-            /* check that the sting in the config file is correctly specified */
-            if (stream_spec != CONFIG_SPECIFIER_UNDEFINED) {
-                SCLogError(SC_ERR_NAPATECH_PARSE_CONFIG,
-                        "Only one Napatech port specifier type allowed.");
-                exit(EXIT_FAILURE);
-            }
-            stream_spec = CONFIG_SPECIFIER_RANGE;
+        if (NapatechUseHWBypass()) {
+#ifdef NAPATECH_ENABLE_BYPASS
+            if (strchr(port->val, '-')) {
+                stream_spec = CONFIG_SPECIFIER_RANGE;
 
-            snprintf(ports_spec, sizeof(ports_spec), "all");
-        } else if (strchr(port->val, '-')) {
-            /* check that the sting in the config file is correctly specified */
-            if (stream_spec != CONFIG_SPECIFIER_UNDEFINED) {
-                SCLogError(SC_ERR_NAPATECH_PARSE_CONFIG,
-                        "Only one Napatech port specifier type allowed.");
-                exit(EXIT_FAILURE);
-            }
-            stream_spec = CONFIG_SPECIFIER_RANGE;
+                char copystr[16];
+                strlcpy(copystr, port->val, sizeof (copystr));
 
-            char copystr[16];
-            strlcpy(copystr, port->val, sizeof(copystr));
+                ports_spec.first[iteration] = atoi(copystr);
```

Comment:
Should the port values be validated against a range of integer values?

Could you use `ByteExtractStringUint{8,16}` instead of `atoi`?